syntax = "proto3";

service Hello {
  rpc SayHello(HelloRequest) returns (HelloResponse);
  rpc SayHelloResponseStream(HelloRequest) returns (stream HelloResponse);
  rpc SayHelloRequestStream(stream HelloRequest) returns (HelloResponse);
  rpc SayHelloBidirectionalStream(stream HelloRequest) returns (stream HelloResponse);
}

message HelloRequest {
  string name = 1;
}

message HelloResponse {
  string name = 1;
  int32 age = 2;
}